### PR TITLE
Minor improvements to docs 

### PR DESCRIPTION
**Pull request by hadarhg**
- Using the Documentation 
- WordCount
- Mobile Gaming
- Java SDK page
- Python SDK page: minor update

Small fix in Resources page

### CODE REVIEW

**Comment by hadarhg** [comment]
@davorbonaci

File: src/documentation/index.md [hunk 1]
```diff
@@ -9,7 +9,7 @@ redirect_from:
 
 # Apache Beam Documentation
 
-Get in-depth conceptual information and reference material for the Beam Model, SDKs and Runners:
+This secton provides in-depth conceptual information and reference material for the Beam Model, SDKs, and Runners:
```
Comment (davorbonaci) [inline src/documentation/index.md]: section

File: src/documentation/index.md [hunk 2]
```diff
@@ -18,6 +18,12 @@ Learn about the Beam Programming Model and the concepts common to all Beam SDKs
 * The [Programming Guide]({{ site.baseurl }}/documentation/programming-guide/) introduces all the key Beam concepts.
 * Visit [Additional Resources]({{ site.baseurl }}/documentation/resources/) for some of our favorite articles and talks about Beam. 
 
+## Pipeline Fundamentals
+
+* [Design Your Pipeline]({{ site.baseurl }}/documentation/pipelines/design-your-pipeline) - Plan your pipeline’s structure, choose transforms to apply to your data, and determine your input and output methods.
```
Comment (davorbonaci) [inline src/documentation/index.md]: / at the end of the links

File: src/documentation/index.md [hunk 3]
```diff
@@ -18,6 +18,12 @@ Learn about the Beam Programming Model and the concepts common to all Beam SDKs
...
+* [Create Your Pipeline]({{ site.baseurl }}/documentation/pipelines/create-your-pipeline) - Learn the mechanics of using the Beam SDKs to build a pipeline. Construct a pipeline by using the classes in the Beam SDKs while following a series of steps.
+* [Test Your Pipeline]({{ site.baseurl }}/documentation/pipelines/test-your-pipeline) - Perform local unit testing on your pipeline code to minimize debugging a pipeline’s remote execution.
```
Comment (davorbonaci) [inline src/documentation/index.md]: make these fragments a sentence?

File: src/documentation/sdks/java.md [hunk 4]
```diff
@@ -6,8 +6,28 @@ redirect_from: /learn/sdks/java/
...
 
-This page is under construction ([BEAM-504](https://issues.apache.org/jira/browse/BEAM-504)).
+The Java SDK for Apache Beam provides a simple, powerful API for building both batch and streaming parallel data processing pipelines in Java. 
+
+
+## Get Started with the Java SDK
+
+Get started with the [Beam Programming Model]({{ site.baseurl }}/learn/programming-guide) to learn the basic concepts that apply to all SDKs in Beam.
+
+See the [Java API Reference]({{ site.baseurl }}/learn/sdks/javadoc/) for more information on indicidual APIs.
```
Comment (davorbonaci) [inline src/documentation/sdks/java.md]: invididual

File: src/documentation/sdks/java.md [hunk 5]
```diff
@@ -6,8 +6,28 @@ redirect_from: /learn/sdks/java/
...
+
+
+## Supported Features
+
+The Java SDK supports all features currently supported by the Beam model.
+
+
+## Supported IO Connectors
+
+* Apache Hadoop HDFS
```
Comment (davorbonaci) [inline src/documentation/sdks/java.md]: there's more, I think. GCP ones, for example.

File: src/documentation/sdks/java.md [hunk 6]
```diff
@@ -6,8 +6,31 @@ redirect_from: /learn/sdks/java/
...
```
Comment (davorbonaci) [inline src/documentation/sdks/java.md]: Apache Hadoop's `FileInputFormat` in Hadoop Distributed File System (HDFS)

File: src/documentation/sdks/java.md [hunk 7]
```diff
@@ -6,8 +6,31 @@ redirect_from: /learn/sdks/java/
...
+* JDBC
```
Comment (davorbonaci) [inline src/documentation/sdks/java.md]: Java Database Connectivity (JDBC)

File: src/documentation/sdks/java.md [hunk 8]
```diff
@@ -6,8 +6,31 @@ redirect_from: /learn/sdks/java/
...
+* JMS
```
Comment (davorbonaci) [inline src/documentation/sdks/java.md]: Java Message Service (JMS)

File: src/documentation/sdks/java.md [hunk 9]
```diff
@@ -6,8 +6,31 @@ redirect_from: /learn/sdks/java/
...
+* Apache Kafka
+* Kinesis
```
Comment (davorbonaci) [inline src/documentation/sdks/java.md]: Amazon Kinesis

File: src/documentation/sdks/java.md [hunk 10]
```diff
@@ -6,8 +6,31 @@ redirect_from: /learn/sdks/java/
...
+* MongoDB
+* Google Cloud BigQuery
```
Comment (davorbonaci) [inline src/documentation/sdks/java.md]: Google BigQuery

File: src/documentation/sdks/java.md [hunk 11]
```diff
@@ -6,8 +6,31 @@ redirect_from: /learn/sdks/java/
...
+* Google Cloud BigTable
++ Google Cloud Datastore
```
Comment (davorbonaci) [inline src/documentation/sdks/java.md]: Additionally, avro, text, xml, pubsub

File: src/documentation/sdks/java.md [hunk 12]
```diff
@@ -6,8 +6,31 @@ redirect_from: /learn/sdks/java/
...
+
+See the [Java API Reference]({{ site.baseurl }}/learn/sdks/javadoc/) for more information on individual APIs.
+
+
+## Supported Features
+
+The Java SDK supports all features currently supported by the Beam model.
+
+
+## Supported IO Connectors
```
Comment (davorbonaci) [inline src/documentation/sdks/java.md]: sort alphabetically

**Comment by hadarhg** [comment]
@davorbonaci Thanks for the comments! Fixed, PTAL.

File: src/get-started/wordcount-example.md [hunk 13]
```diff
@@ -74,7 +72,7 @@ The Minimal WordCount pipeline contains several transforms to read data into the
 
 Each transform takes some kind of input (data or otherwise), and produces some output data. The input and output data is represented by the SDK class `PCollection`. `PCollection` is a special class, provided by the Beam SDK, that you can use to represent a data set of virtually any size, including infinite data sets.
 
-<img src="{{ "/images/wordcount-pipeline.png" | prepend: site.baseurl }} alt="Word Count pipeline diagram"">
+<img src="{{ "/images/wordcount-pipeline.png" | prepend: site.baseurl }} alt="Word Count pipeline diagram">
```
Comment (davorbonaci) [inline src/get-started/wordcount-example.md]: quote should be after `}}`, and before `alt`, right?

File: src/documentation/sdks/java.md [hunk 14]
```diff
@@ -6,8 +6,36 @@ redirect_from: /learn/sdks/java/
...
+
+* Amazon Kinesis
+* Apache Hadoop's `FileInputFormat` in Hadoop Distributed File System (HDFS)
+* Apache Kafka
+* Avro Files
+* Google BigQuery
+* Google Cloud BigTable
```
Comment (davorbonaci) [inline src/documentation/sdks/java.md]: Bigtable

File: src/get-started/mobile-gaming-example.md [hunk 15]
```diff
@@ -21,9 +21,18 @@ Every time a user plays an instance of our hypothetical mobile game, they genera
...
 
-This means that some data events might be received by the game server significantly later than users generate them. This time difference can have processing implications for pipelines that make calculations that consider when each score was generated. Such pipelines might track scores generated during each hour of a day, for example, or they calculate the length of time that users are continuously playing the game—both of which depend on each data record's event time.
+The following diagram shows the ideal situation vs reality. The X-axis represents event time: the actual time a game even occurred. The Y-axis represents processing time: the time at which a game event was processed. Ideally, events should be processed as they occur, depicted by the dotted line in the diagram. However, in reality that is not the case and reality looks more like what is depicted by the red squiggly line.
```
Comment (davorbonaci) [inline src/get-started/mobile-gaming-example.md]: a game even -> a user action

**Pull request closed without merging**